ecommerce/models.py

```diff
@@ -154,3 +154,53 @@ def save(self, *args, **kwargs):
     def __str__(self):
         """Description for CoursePrice"""
         return "CoursePrice for {}, price={}, is_valid={}".format(self.course_run, self.price, self.is_valid)
+
+
+class Coupon(Model):
+    """
+    Model for coupons representing discounts on purchases
+    """
+    CREATED = 'created'
+    USED = 'used'
+    DELETED = 'deleted'
+    STATUSES = [CREATED, USED, DELETED]
+
+    PERCENT_DISCOUNT = 'percent-discount'
+    FIXED_DISCOUNT = 'fixed-discount'
+
+    AMOUNT_TYPES = [PERCENT_DISCOUNT, FIXED_DISCOUNT]
+
+    COURSE_RUN = 'course-run'
+    COURSE = 'course'
+    PROGRAM = 'program'
+    PRODUCT_TYPES = [PROGRAM, COURSE, COURSE_RUN]
+
+    order = ForeignKey(Order, null=True)
```

Comment:
I think we want coupons to apply to program, course, or course run, but you're right that this should be line in any case